kafka/producer/base.py

```diff
@@ -76,6 +77,20 @@ def _send_upstream(queue, client, codec, batch_time, batch_size,
             defaults to True.
         stop_timeout (int or float, optional): number of seconds to continue
             retrying messages after stop_event is set, defaults to 30.
+
+    Events emitted (no args unless specified):
+        async.producer.connect.succeed
+        async.producer.connect.fail
+        async.producer.stop(unsent_messages)
+        async.producer.queue.pop(topic_partition, msg, key)
+        async.producer.request.send(requests)
+        async.producer.request.succeed(request)
+        async.producer.request.error(error_cls, request)
+        async.producer.request.retry(request)
+        async.producer.backoff(time_in_ms)
+        async.producer.metadata.refresh
+        async.producer.metadata.refresh.fail
```

Comment:
few things. 1. Where did you get these values from?  grepping?
2. Why are we only adding event_registrar to async events?